packages/text-input/src/TextInput.tsx

```diff
@@ -0,0 +1,328 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { css, cx } from '@leafygreen-ui/emotion';
+import Icon from '@leafygreen-ui/icon';
+import { uiColors } from '@leafygreen-ui/palette';
+import { useUsingKeyboardContext } from '@leafygreen-ui/leafygreen-provider';
+import { createDataProp } from '@leafygreen-ui/lib';
+
+export const State = {
+  none: 'none',
+  valid: 'valid',
+  error: 'error',
+} as const;
+
+export type State = typeof State[keyof typeof State];
+
+interface TextInputProps {
+  /**
+   * Text shown in bold above the input element.
+   */
+  label?: string;
+  /**
+   * Text that gives more detail about the requirements for the input.
+   */
+  description?: string;
+  /**
+   * Whether or not the field is optional. This is false by default.
+   */
+  optional?: boolean;
+  /**
+   * Whether or not the field is currently disabled. This is false by default.
+   */
+  disabled?: boolean;
+  /**
+   * Callback to be executed when the value of the input field changes.
+   */
+  onChange?: Function;
+  /**
+   * The placeholder text shown in the input field before the user begins typing.
+   */
+  placeholder?: string;
+  /**
+   * The message shown below the input field if the value is invalid.
+   */
+  errorMessage?: string;
+  /**
+   * The current state of the TextInput. This can be none, valid, or error.
+   */
+  state?: State;
+  /**
+   * The current value of the input field. If a value is passed to this prop, component will be controlled by consumer.
+   */
+  value?: string;
+  /**
+   * Callback used to set the value of the input field.
+   */
+  setValue?: Function;
+  /**
+   * className supplied to the TextInput container.
+   */
+  className?: string;
+}
+
+/**
+ * # TextInput
+ *
+ * TextInput component
+ *
+ * ```
+<TextInput label={'Input Label'} onChange={() => execute when value of input field changes}/>
```

Comment:
Don't need brackets around string for label

Reply:
Fixed